diff -urN 1.0.x/samsdaemon.c build/samsdaemon.c
--- 1.0.x/samsdaemon.c	2007-12-15 10:19:50.000000000 +0400
+++ build/samsdaemon.c	2007-12-15 10:21:19.000000000 +0400
@@ -2583,7 +2583,7 @@
      printf("Ok \n");
 
   
-  strcpy(&squiduser[0],"squid");
+  strcpy(&squiduser[0],"proxy");
   i=0;
   tt=time(NULL);
   t=localtime(&tt);
@@ -2956,10 +2956,10 @@
 	         {
                     sprintf(&str[0],"squidGuard -C all -c %s/squidGuard.conf",conf.sgdbpath);
                     system(&str[0]);
-                    sprintf(&str[0],"chown nobody:nobody %s/_sams_*",conf.sgdbpath);
+                    sprintf(&str[0],"chown nobody:nogroup %s/_sams_*",conf.sgdbpath);
                     system(&str[0]);
                     sprintf(&str[0],"chown -R %s %s/_sams_*", &squiduser[0], conf.sgdbpath);
-		    system(&str[0]);
+                    system(&str[0]);
 	         }
                /* Задаем права доступа на каталог с Режиком */
                if(RREJIK==1)
